creator-node/src/services/stateMachineManager/stateMachineUtils.js

```diff
@@ -267,56 +267,63 @@ const makeMetricToRecord = (
 
 const makeQueue = ({
   name,
+  processor,
+  logger,
   removeOnComplete,
   removeOnFail,
-  lockDuration,
-  prometheusRegistry = null,
-  limiter = null
+  prometheusRegistry,
+  concurrency = 1,
+  limiter = null,
+  onFailCallback = null
 }) => {
-  // Settings config from https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#advanced-settings
-  const queue = new BullQueue(name, {
-    redis: {
-      host: config.get('redisHost'),
-      port: config.get('redisPort')
-    },
+  const connection = {
+    host: config.get('redisHost'),
+    port: config.get('redisPort')
+  }
+  const queue = new Queue(name, {
+    connection,
     defaultJobOptions: {
       removeOnComplete,
       removeOnFail
-    },
-    settings: {
-      // Should be sufficiently larger than expected job runtime
-      lockDuration,
-      // We never want to re-process stalled jobs
-      maxStalledCount: 0
-    },
+    }
+  })
+
+  const worker = new Worker(name, processor, {
+    connection,
+    concurrency,
     limiter
   })
+  if (limiter) {
+    const scheduler = new QueueScheduler(name, { connection })
+  }
+
+  _registerQueueEvents(worker, logger)
+  queue.on(
+    'failed',
+    onFailCallback ||
+      ((job, error, prev) => {
+        const loggerWithId = createChildLogger(logger, {
+          jobId: job?.id || 'unknown'
+        })
+        loggerWithId.error(
+          `Job failed to complete. ID=${job?.id}. Error=${error}`
+        )
+      })
+  )
 
   if (prometheusRegistry !== null && prometheusRegistry !== undefined) {
-    prometheusRegistry.startQueueMetrics(queue)
+    prometheusRegistry.startQueueMetrics(queue, worker)
   }
 
-  return queue
+  return { queue, worker, logger }
 }
 
-const registerQueueEvents = (queue, queueLogger) => {
-  queue.on('global:waiting', (jobId) => {
-    const logger = createChildLogger(queueLogger, { jobId })
-    logger.info('Job waiting')
-  })
-  queue.on('global:active', (jobId, jobPromise) => {
-    const logger = createChildLogger(queueLogger, { jobId })
+const _registerQueueEvents = (worker, queueLogger) => {
+  worker.on('active', (job, prev) => {
+    const logger = createChildLogger(queueLogger, { jobId: job.id })
     logger.info('Job active')
   })
-  queue.on('global:lock-extension-failed', (jobId, err) => {
-    const logger = createChildLogger(queueLogger, { jobId })
-    logger.error(`Job lock extension failed. Error: ${err}`)
-  })
-  queue.on('global:stalled', (jobId) => {
-    const logger = createChildLogger(queueLogger, { jobId })
-    logger.error('Job stalled')
-  })
-  queue.on('global:error', (error) => {
+  worker.on('error', (error) => {
```

Comment:
agreed on not logging waiting job, but what about lock-extension-failed and stalled?

Reply:
I added back stalled, but BullMQ removed lock-extension-failed. [this is the new list](https://api.docs.bullmq.io/interfaces/WorkerListener.html)